ckanext/fork/helpers.py

```diff
@@ -60,3 +64,33 @@ def fork_metadata(resource):
 
     return metadata
 
+
+def get_current_resource(context, resource):
+    try:
+        current = toolkit.get_action('resource_show')(
+            context,
+            {"id": resource["id"]}
+        )
+    except logic.NotFound as e:
+        log.info(f"No existing resource found with error: {e}")
+        current = {}
+    except KeyError as e:
+        log.info(f"No resource id provided, probably a new resource. Original error: {e}")
+        current = {}
+    except Exception as e:
+        log.exception(f"I can't find the original resource for unknown reason: {e}")
+        raise e
```

Comment:
I've proposed a suggestion below based on 3 thoughts:

1. Are we confident that our log messages are expressed in the best possible form? Especially unsure of the phrase "No existing resource found with error: {e}" 

2. Is it possible that a key error be raised for a different reason than the one you are catching, and that this could break your code? Should you not check a resource ID is provided before you act, rather than catch the exception after you have acted?

3. Also - does the third `except` really add any value?  You are just catching "all other exceptions" and re raising them with "I can't find the original resource", but does this extra log really add any value?  

```suggestion
def get_current_resource(context, resource):
    current_resource = {}
    if resource.get("id"):
        try:
            current_resource = toolkit.get_action('resource_show')(
                context,
                {"id": resource["id"]}
            )
        except logic.NotFound as e:
            log.info(f"Resource {resource['id']} does not exist "
                     "- must be creating a new resource with specific id.")
    return current_resource
```

Reply:
Hmmm... I am not sure but I think this change will always be flagged as changing on any field that is present in the incoming resource information? I'll implement it and run the new test suite :tada: :wink:

Reply:
Ah, well the suite passed. I'll just do everything manually too as these are new tests and if they pass will accept the change!

Reply:
Looks good!